src/XMakeTasks/WriteCodeFragment.cs

```diff
@@ -275,5 +280,134 @@ private string GenerateCode(out string extension)
             // as there's no point writing the file
             return haveGeneratedContent ? code : String.Empty;
         }
+#endif
+
+        /// <summary>
+        /// Generates the code into a string.
+        /// If it fails, logs an error and returns null.
+        /// If no meaningful code is generated, returns empty string.
+        /// Returns the default language extension as an out parameter.
+        /// </summary>
+        private string GenerateCodeCoreClr(out string extension)
```

Comment:
I was going to do that, but I did it this way since the other method will compile against both configurations, it was easier to test that way. No strong preference.